### PR TITLE
only emit when things change

### PR DESCRIPTION
**Pull request by MattDHill**

### CODE REVIEW

File: frontend/projects/ui/src/app/pages/apps-routes/app-list/package-info.pipe.ts [hunk 1]
```diff
@@ -18,12 +15,12 @@ export class PackageInfoPipe implements PipeTransform {
     private readonly depErrorService: DepErrorService,
   ) {}
 
-  transform(pkg: PackageDataEntry): Observable<PkgInfo> {
+  transform(pkgId: string): Observable<PkgInfo> {
+    firstValueFrom(this.patch.watch$('package-data', pkgId)).then(console.error)
```
Comment (dr-bonez) [inline frontend/projects/ui/src/app/pages/apps-routes/app-list/package-info.pipe.ts]: this looks like debug info. remove?

**Review by dr-bonez — DISMISSED** [review]

**Review by Blu-J — DISMISSED** [review]

File: frontend/projects/ui/src/app/services/dep-error.service.ts [hunk 2]
```diff
@@ -26,13 +34,17 @@ export class DepErrorService {
...
-          (errors, { id }): PackageDependencyErrors => ({
+          (errors, { id }): AllDependencyErrors => ({
             ...errors,
             [id]: this.getDepErrors(pkgs, id, errors),
           }),
-          {} as PackageDependencyErrors,
+          {} as AllDependencyErrors,
         ),
     ),
+    startWith({} as AllDependencyErrors),
```
Comment (Blu-J) [inline frontend/projects/ui/src/app/services/dep-error.service.ts:44]: Aren't these 44-47 just [distinctUntilChanged](https://rxjs.dev/api/operators/distinctUntilChanged)?

Comment (Blu-J) [inline frontend/projects/ui/src/app/services/dep-error.service.ts:44]:
```suggestion
    distinctUntilChanged(deepEqual),
```

File: frontend/projects/ui/src/app/services/dep-error.service.ts [hunk 3]
```diff
@@ -41,29 +53,40 @@ export class DepErrorService {
     private readonly patch: PatchDB<DataModel>,
   ) {}
 
+  getPkgDepErrors$(pkgId: string) {
+    return this.depErrors$.pipe(
+      map(depErrors => depErrors[pkgId]),
+      startWith({} as PkgDependencyErrors),
+      pairwise(),
+      filter(([prev, curr]) => !deepEqual(prev, curr)),
+      map(([_, curr]) => curr),
```
Comment (Blu-J) [inline frontend/projects/ui/src/app/services/dep-error.service.ts:62]:
```suggestion
      map(depErrors => depErrors[pkgId]),
      distinctUntilChanged(deepEqual)
```

**Review by Blu-J — APPROVED** [review]

**Pull request merged**